src/main/scala/li/cil/oc/server/component/Trade.scala

```diff
@@ -0,0 +1,256 @@
+package li.cil.oc.server.component
+
+import java.util.UUID
+
+import li.cil.oc.Settings
+import li.cil.oc.api.machine._
+import li.cil.oc.api.prefab.AbstractValue
+import li.cil.oc.common.EventHandler
+import li.cil.oc.util.InventoryUtils
+import net.minecraft.entity.passive.EntityVillager
+import net.minecraft.item.ItemStack
+import net.minecraft.nbt.NBTTagCompound
+import net.minecraft.village.MerchantRecipe
+import net.minecraftforge.common.DimensionManager
+import net.minecraftforge.common.util.ForgeDirection
+import ref.WeakReference
+import li.cil.oc.api.network.EnvironmentHost
+import scala.collection.JavaConverters._
+
+class Trade(info: TradeInfo) extends AbstractValue {
+
+  def this() = this(new TradeInfo())
+  def this(upgr: UpgradeTrading, villager: EntityVillager, recipeID: Int) = {
+    this(new TradeInfo(upgr.host, villager, recipeID))
+  }
+
+
+  override def load(nbt: NBTTagCompound) = {
+    EventHandler.scheduleServer(() => {
+      //Tell the info to load from NBT, behind EventHandler because when load is called we can't access the world yet
+      //and we need to access it to get the Robot/Drone TileEntity/Entity
+      info.load(nbt)
+    })
+  }
+
+  override def save(nbt: NBTTagCompound) = {
+    //Tell info to save to nbt
+    info.save(nbt)
+  }
+
+  def inventory = info.inventory
+
+  @Callback(doc="function():table, table -- returns the items the villager wants for this trade")
+  def getInput(context: Context, arguments: Arguments): Array[AnyRef] = {
+    Array(info.recipe.getItemToBuy.copy(),
+      info.recipe.hasSecondItemToBuy match {
+        case true => info.recipe.getSecondItemToBuy.copy()
+        case false => null
+      })
+  }
+
+  @Callback(doc = "function():table -- returns the item the villager offers for this trade")
+  def getOutput(context: Context, arguments: Arguments): Array[AnyRef] = {
+    Array(info.recipe.getItemToSell.copy())
+  }
+
+  @Callback(doc="function():boolean -- returns whether the villager currently wants to trade this")
+  def isEnabled(context: Context, arguments: Arguments): Array[AnyRef] = {
+    Array(info.villager.exists((villager: EntityVillager) => // Make sure villager is neither dead/gone nor the recipe
+      !info.recipe.isRecipeDisabled                          // has been disabled
+    ).asInstanceOf[AnyRef])
```

Comment:
Also for future reference, [ResultWrapper.result](https://github.com/MightyPirates/OpenComputers/blob/master-MC1.7.10/src/main/scala/li/cil/oc/util/ResultWrapper.scala) instead of Array, to save yourself the hassle of manually boxing all the time.